### DISCUSSION TITLE
Threading performance issue + possible desync when changing threading mid-map

### DISCUSSION THREAD

**Discussion by MaxKruse** [ppy/osu#18694] [Q&A]
### Type

Performance

### Bug description

Installed lazer today again and tries playing.

Assuming Multithreading is the preferred method (as it usually is in most applications or games, and no other hint to mention this, as well as being the default option), Performance significantly tanks when enabling multi thread.

Expectation: Multi Thread is both smoother and possibly more performant.
Reality: 

Multi Thread starves the render Threa. For Multithread, 1000Hz is displayed, but 500fps rendered "lag" is notcable.  
For Single thread, no Hz is displayed (as expected), but render Performance finally catches up to the framelimit set (e.g. x4 and x8, as x2 is noticably stuttery for me). 

Additionally, changing this setting mid gameplay causes noticable audio delay, as seen in the video linked below.

For my relevant hardware:

Windows 10 Build 19043.1741 (21H1)
AMD R5 5600x (Stock)
AMD RX 6700xt (Stock)

Additionally, regardless of the rendering method used, frequent stutters are present.

### Screenshots or videos

Note: This behaviour is exactly the same, regardless of having a capture running (which was done with the GPU Encoder in this case), or regular gameplay. I have double-checked the occurances of the noticable problems.

https://youtu.be/yx1hNfMh3Hw

The noticable occurances are as followed (rough timestamps):
00:41 Singlethread Stutter
01:20 Multithread Stutter
01:40 Audio Shift noticable from here on out. (After changing back to Singlethread

### Version

2022.612.0-lazer

### Logs

[legacy-ipc.log](https://github.com/ppy/osu/files/8886253/legacy-ipc.log)
[network.log](https://github.com/ppy/osu/files/8886254/network.log)
[performance.log](https://github.com/ppy/osu/files/8886255/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8886256/runtime.log)
[updater.log](https://github.com/ppy/osu/files/8886257/updater.log)
[database.log](https://github.com/ppy/osu/files/8886258/database.log)

**Comment by bdach** [comment]
please link the beatmap this was reproduced on, and recapture the "stutters" on videos where the frame graph is showing (press ctrl-f11 twice).

**Comment by MaxKruse** [comment]
Beatmap in question is https://osu.ppy.sh/beatmapsets/799156#osu/1771787

Updated Video: https://youtu.be/iPxKqx2f6Oc

Updated Logs:
[performance.log](https://github.com/ppy/osu/files/8888459/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8888460/runtime.log)
[updater.log](https://github.com/ppy/osu/files/8888461/updater.log)
[database.log](https://github.com/ppy/osu/files/8888462/database.log)
[legacy-ipc.log](https://github.com/ppy/osu/files/8888463/legacy-ipc.log)
[network.log](https://github.com/ppy/osu/files/8888464/network.log)

**Comment by peppy** [comment]
May be the same issue as reported over at https://github.com/ppy/osu/issues/18652.

@MaxKruse could you try the instructions provided in that thread to test against the last release and see if you still get the issue?

**Comment by MaxKruse** [comment]
As mentioned in #18652 The audio-issue seems to not occur in 605.1

The underlying threading performance issue is however persistent across versions. Possibly linked issues, due to threading performance?

**Comment by frenzibyte** [comment]
Definitely not linked if one doesn't occur in previous release while the other does. Going to also move this to Q&A given that it's not a direct regression from previous release.
